stdlib/Artifacts/src/Artifacts.jl

```diff
@@ -386,18 +400,19 @@ function artifact_meta(name::String, artifact_dict::Dict, artifacts_toml::String
     if isa(meta, Vector)
         dl_dict = Dict{AbstractPlatform,Dict{String,Any}}()
         for x in meta
-            x::Dict{String}
-            dl_dict[unpack_platform(x, name, artifacts_toml)] = x
+            y = x::Dict{String}
+            dl_dict[unpack_platform(y, name, artifacts_toml)] = y
         end
         meta = select_platform(dl_dict, platform)
+    end
+    meta === nothing && return nothing
     # If it's NOT a dict, complain
-    elseif !isa(meta, Dict)
+    if !isa(meta, Dict{String})
```

Comment:
this makes the `isa` check more expensive, since we have a fast path for unparameterized types, but it cannot optimize partially parameterized times